sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/export/trace/_exporter.py

```diff
@@ -162,76 +176,142 @@ def _convert_span_to_envelope(span: Span) -> TelemetryItem:
-                    data.target = "{}/{}".format(
-                        span.attributes["net.peer.ip"],
-                        span.attributes["messaging.destination"]
-                    )
-                else:
-                    data.target = span.attributes["messaging.system"]
+                # TODO: data.data for rpc
+                # rpc specific logic for target
+                if "peer.service" not in span.attributes:
+                    target = span.attributes["rpc.system"]
             else:
                 # TODO: Azure specific types
                 data.type = "N/A"
+        elif span.kind is SpanKind.PRODUCER:  # Messaging
+            data.type = "Queue Message"
+            # TODO: data.data for messaging
+            # TODO: Special logic for data.target for messaging?
         else:  # SpanKind.INTERNAL
             data.type = "InProc"
             data.success = True
-    for key in span.attributes:
+        # Apply truncation
+        if data.result_code:
+            data.result_code = data.result_code[:1024]
+        if data.data:
+            data.data = data.data[:8192]
+        if target:
+            data.target = target[:1024]
+    for key, val in span.attributes.items():
         # Remove Opentelemetry related span attributes from custom dimensions
         if key.startswith("http.") or \
                 key.startswith("db.") or \
                 key.startswith("rpc.") or \
                 key.startswith("net.") or \
                 key.startswith("messaging."):
             continue
-        data.properties[key] = span.attributes[key]
+        # Apply truncation rules
+        # Max key length is 150, value is 8192
+        if not key or len(key) > 150 or val is None:
+            continue
+        data.properties[key] = val[:8192]
     if span.links:
+        # Max length for value is 8192
+        # Since links are a fixed length (80) in json, max number of links would be 102
         links = []
         for link in span.links:
+            if len(links) > 102:
+                break
             operation_id = "{:032x}".format(link.context.trace_id)
             span_id = "{:016x}".format(link.context.span_id)
             links.append({"operation_Id": operation_id, "id": span_id})
         data.properties["_MS.links"] = json.dumps(links)
-    # TODO: tracestate, tags
     return envelope
+
+
+def _get_default_port_db(dbsystem):
+    if dbsystem == "postgresql":
+        return 5432
+    if dbsystem == "mysql":
+        return 3306
+    if dbsystem == "memcached":
+        return 11211
+    if dbsystem == "mongodb":
+        return 27017
+    if dbsystem == "redis":
+        return 6379
+    return 0
+
+
+def _get_default_port_http(scheme):
+    if scheme == "http":
+        return 80
+    if scheme == "https":
+        return 443
+    return 0
+
+
+def _is_relational_db(dbsystem):
+    return dbsystem in ["postgresql", "mysql"]
```

Comment:
There are way more relational DBs available in this property https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/database.md#connection-level-attributes do we want to add this "SQL" type, is kidna weird that we do not use the dbSystem always if available

Reply:
Ok, I will follow up on spec discussions, having this kind of check seems unnecessary unless Breeze do something specific when type = "SQL"